### DISCUSSION TITLE
npx sb extract gives timeout error

### DISCUSSION THREAD

**Discussion by TomWebbio** [storybookjs/storybook#15141] [Help] [answered]
The `npx sb extract` command gives the following error:
`TimeoutError: waiting for function failed: timeout 30000ms exceeded`

This happens every time. It can be easily reproduced by creating a new Project using CRA (for example),  running `npx sb init` to get started with a basic Storybook setup and then trying to generate the documentation using `npx sb extract`.

Could anyone please let me know if there's any way for me to get this to work? I've tried various NPM versions but I doubt that has anything to do with the issue.

**Comment by ndelangen** [comment, marked as answer]
@TomWebbio You have to give the sb extract command to location or url of where storybook is to be found, and where to place stories.json:

Where it's hosted:
```sh
npx sb extract http://localhost:6006/iframe.html ./stories.json
```

Where a storybook-static is located:
```sh
npx sb extract ./storybook-static
```

We're supposedly checking if these values are empty:
https://github.com/storybookjs/storybook/blob/736389950a299a76f948490b73fc78ae99130874/lib/cli/src/extract.ts#L70-L82

But maybe we're passing in default values in there?

yeah we are:
https://github.com/storybookjs/storybook/blob/194a6991047259762fe9a904c9905c3610a3796a/lib/cli/src/generate.ts#L86-L94

But we're never checking if these actually exist... that's the issue.

The user isn't specifying anything, it's using the default options, but the source directory doesn't exist...
This results in an page hosted that's un-extractable.. thus it times out.

**Reply by TomWebbio** [reply]
Hi Norbert,

Thanks for the quick reply. Adding those parameters did the trick and the stories.json file is indeed generated. I wasn't able to find these required parameters in the documentation, but if they are documented then consider it an oversight on my part. 

I will continue working on getting this to work. My endgoal is getting storybook composition to work, and although the stories.json file is marked as optional, it gives an error in the console when it cannot find it.

If I can't figure it out I will make a separate QA thread. Thanks agaiin for your help.

**Reply by ndelangen** [reply]
Yeah, the error in the console is a network request failure, not a storybook failure (AFAIK).
